Google.Api.Gax.Grpc/GrpcCoreAdapter.cs

```diff
@@ -0,0 +1,165 @@
+﻿/*
+ * Copyright 2020 Google LLC
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using Grpc.Core;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Threading;
+
+namespace Google.Api.Gax.Grpc
+{
+    /// <summary>
+    /// Implementation of <see cref="GrpcAdapter"/> for Grpc.Core.
+    /// </summary>
+    public sealed class GrpcCoreAdapter : GrpcAdapter
+    {
+        private const string GrpcCoreAssemblyName = "Grpc.Core, Version=2.0.0.0, Culture=neutral, PublicKeyToken=d754f35622e28bad";
+
+        // Option names, copied from ChannelOptions
```

Comment:
It's a little weird that we have these harcoded here (ultimately a dependency). I don't mind at all tbh, it just jumped at me a little be.

I know it will be more reflection, but maybe use the `Grpc.Core.ChannelOptions` class with all the constants? That way, in the unlikely case those change, we don't have to change the code.

Reply:
I'd prefer not to - to be honest, I think it's more likely that the names in the C# code will change e.g. in a new major version than that the underlying values will change, given that the whole ecosystem will depend on those.